### PR TITLE
Adding links to PTransform Colab Notebook.

### PR DESCRIPTION
**Pull request by svetakvsundhar**
Addresses #26511 temporarily.

TODO: Embed colab tutorial on the webpage itself.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by svetakvsundhar** [comment]
R: @tvalentyn 

Note that we will link this colab notebook for the tutorial for now, incoming change to embed colab cells into the website itself.

File: website/www/site/content/en/documentation/programming-guide.md [hunk 1]
```diff
@@ -160,6 +160,10 @@ Pipeline p = Pipeline.create(options);
 {{< code_sample "sdks/typescript/test/docs/programming_guide.ts" pipelines_constructing_creating >}}
 {{< /highlight >}}
 
+For a more in-depth tutorial of applying various transforms
```
Comment (tvalentyn) [inline website/www/site/content/en/documentation/programming-guide.md:163]: noting that the intro text is the same but the notebook link is different. is that intentional?

**Comment by tvalentyn** [comment]
PTAL at website errors (https://github.com/apache/beam/actions/runs/4918577372/jobs/8785092715?pr=26589).

**Comment by tvalentyn** [comment]
actually website looks like be a preexisting error as per https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/

File: website/www/site/content/en/documentation/programming-guide.md (hunk 1 above)
Comment (svetakvsundhar) [reply website/www/site/content/en/documentation/programming-guide.md]: It was intentional, but probably better to distinguish the terminology. See new commit

**Comment by tvalentyn** [comment]
Run Website PreCommit

**Review by tvalentyn — APPROVED** [review]

**Pull request merged**